tests/api2/test_190_filesystem.py

```diff
@@ -95,238 +69,194 @@ def test_05_set_immutable_flag_on_path(request):
-        {'tag': 'group@', 'id': -1, 'type': 'ALLOW', 'perms': {'BASIC': 'FULL_CONTROL'}, 'flags': {'BASIC': 'INHERIT'}},
-        {'tag': 'USER', 'id': 65534, 'type': 'ALLOW', 'perms': {'BASIC': 'FULL_CONTROL'}, 'flags': {'BASIC': 'INHERIT'}},
-        {'tag': 'GROUP', 'id': 65534, 'type': 'ALLOW', 'perms': {'BASIC': 'FULL_CONTROL'}, 'flags': {'BASIC': 'INHERIT'}},
+        {'tag': 'owner@', 'id': -1, 'type': 'ALLOW', 'perms': perm, 'flags': flag},
+        {'tag': 'group@', 'id': -1, 'type': 'ALLOW', 'perms': perm, 'flags': flag},
+        {'tag': 'USER', 'id': 65534, 'type': 'ALLOW', 'perms': perm, 'flags': flag},
+        {'tag': 'GROUP', 'id': 65534, 'type': 'ALLOW', 'perms': perm, 'flags': flag},
     ]
-
-    # Do NFSv4 checks
-    with create_dataset(target, {'acltype': 'NFSV4', 'aclmode': 'PASSTHROUGH'}, None, 770):
-        sub_ds_name = f'{ds_name}/sub'
-        sub_target = f'{pool_name}/{sub_ds_name}'
-        sub_path = f'/mnt/{sub_target}'
-
+    with create_dataset(
+        'acl_execute_test',
+        data={'acltype': 'NFSV4', 'aclmode': 'PASSTHROUGH'},
+        mode=770
+    ) as ds:
+        path = f'/mnt/{ds}'
         """
         For NFSv4 ACLs four different tags generate user tokens differently:
         1) owner@ tag will test `uid` from payload
         2) group@ tag will test `gid` from payload
         3) GROUP will test the `id` in payload with id_type
         4) USER will test the `id` in mayload with USER id_type
         """
-
         # Start with testing denials
-        with create_dataset(sub_target, {'acltype': 'NFSV4', 'aclmode': 'PASSTHROUGH'}):
+        with create_dataset(
+            'acl_execute_test/sub',
+            data={'acltype': 'NFSV4', 'aclmode': 'PASSTHROUGH'}
+        ) as sub_ds:
+            sub_path = f'/mnt/{sub_ds}'
             acl = deepcopy(NFSV4_DACL)
-            names = ['daemon', 'apps', 'nobody', 'nogroup']
+            names = ('daemon', 'apps', 'nobody', 'nogroup')
             for idx, entry in enumerate(NFSV4_DACL):
-                perm_job = POST('/filesystem/setacl/',
-                                {'path': sub_path, "dacl": acl, 'uid': 1, 'gid': 568})
-                assert perm_job.status_code == 200, perm_job.text
-
-                job_status = wait_on_job(perm_job.json(), 180)
-
-                # all of these tests should fail
-                assert job_status['state'] == 'FAILED', str(job_status['results'])
-                assert names[idx] in job_status['results']['error'], job_status['results']['error']
+                with pytest.raises(ClientException, match=f'{names[idx]}'):
+                    # all of these tests should fail and the user account
+                    # should be in the error message raised
+                    call(
+                        'filesystem.setacl',
+                        {'path': sub_path, "dacl": acl, 'uid': 1, 'gid': 568},
```

Comment:
Small nit, just the double quotations here. Looks like they were in the old code too lol